src/dlls/dbgshim/dbgshim.cpp

```diff
@@ -387,25 +387,63 @@ class RuntimeStartupHelper
         return hr;
     }
 
+    bool NoNullHandles(HANDLE *handleArray, DWORD arrayLength)
+    {
+        for (int i = 0; i < (int)arrayLength; i++)
+        {
+            HANDLE h = handleArray[i];
+            if (h == NULL)
```

Comment:
I wasn't really sure if it should check for INVALID_HANDLE_VALUE.  It looks to me like the only way it could be INVALID_HANDLE_VALUE if the CreateEvent fails on the runtime side. For that unlikely condition (the runtime has a CONSISTENCY_CHECK for it which I think is DEBUG only), dbgshim should probably fail instead of sleep/retry.